src/views/projects/Issue/IssueSidebar.svelte

```diff
@@ -0,0 +1,210 @@
+
+  export let action: "create" | "edit" | "view" = "view";
+  export let assignees: string[] = [];
+  export let tags: string[] = [];
+
+  const dispatch = createEventDispatcher<{
+    saveAssignees: string[];
+    saveTags: string[];
+  }>();
+
+  let editAssignees: boolean = false;
+  let editTags: boolean = false;
+
+  let assignee: string = "";
+  let assigneeCaption: string | undefined = undefined;
+  let tag: string = "";
+  let tagCaption: string | undefined = undefined;
+
+  function handleAddAssignee() {
+    const nodeId = parseNodeId(assignee);
+    if (nodeId) {
+      if (assignees.includes(nodeId.pubkey)) {
+        assigneeCaption = "This user is already assigned";
+        return;
+      }
+      assignees.push(`${nodeId.prefix}${nodeId.pubkey}`);
+      // Allow component to react to array change.
+      assignees = assignees;
+      if (action === "create") {
+        dispatch("saveAssignees", assignees);
+      }
+      assignee = "";
+    } else {
+      assigneeCaption = "This user is not valid";
+    }
+  }
+
+  function handleAddTag() {
+    if (tags.includes(tag)) {
+      tagCaption = "This tag already exists";
+      return;
+    }
+    tags.push(tag);
+    // Allow component to react to array change.
+    tags = tags;
+    if (action === "create") {
+      dispatch("saveTags", tags);
+    }
+    tag = "";
+  }
+</script>
+
+<style>
+  .assignee {
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+  }
+  .tag {
+    max-width: 15rem;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+  }
+  .metadata {
+    margin-left: 1rem;
+    border-radius: var(--border-radius);
+    font-size: var(--font-size-small);
+    padding-left: 1rem;
+  }
+  .metadata-section {
+    margin-bottom: 1rem;
+    border-bottom: 1px dashed var(--color-foreground-4);
+  }
+  .metadata-section-header {
+    display: flex;
+    gap: 1rem;
+    align-items: center;
+    font-size: var(--font-size-small);
+    margin-bottom: 0.75rem;
+    color: var(--color-foreground-5);
+  }
+  .metadata-section-body {
+    display: flex;
+    flex-wrap: wrap;
+    flex-direction: row;
+    gap: 0.5rem;
+    margin-bottom: 1.25rem;
+  }
+  .metadata-section-empty {
+    color: var(--color-foreground-6);
+  }
+</style>
+
+<div class="metadata layout-desktop">
+  <div class="metadata-section">
+    <div class="metadata-section-header">
+      <span>Assignees</span>
+      {#if action === "edit"}
+        {#if !editAssignees}
+          <Button
+            size="tiny"
+            variant="text"
+            on:click={() => (editAssignees = !editAssignees)}>
+            edit
+          </Button>
+        {:else if editAssignees}
+          <Button
+            size="tiny"
+            variant="text"
+            on:click={() => {
+              dispatch("saveAssignees", assignees);
+              editAssignees = !editAssignees;
+            }}>
+            save
+          </Button>
+        {/if}
```

Comment:
Can we clean up this `if`, it seems like `if editAssignees ... else ...` should do, no?